### PR TITLE
Feature suggestions model

### LINKED ISSUE #106: Remember previous feedback for an exercise
This is the first step towards a new big feature: the ability of the platform to "suggest" feedback to a teacher, when correcting students' assignments. See https://github.com/mumuki/mumuki-platform/issues/275 for more information.

Every time a teacher sends a message it's stored in a suggestions collection, indexed by `(guide slug, exercise id)`, with the following extra data:
* a times used counter, initially 0;
* the exercise's Mongo id;
* the submission that originated it (inside a list). 

Later, when any teacher corrects the same exercise, this message would be suggested. If she uses it, the counter will be increased by 1, the new submission will be added to the list, and the new message will be stored too, if different (because editing will be allowed in the UI). If not, a new message will be stored, as described above.

### Doubts
1. Should/can the existing `Message` and `Assignment` models be reused? I believe the model is strongly tied to the collection it generates, but further research on the matter is pending.
2. How to deal with further editions of a suggestion? For simplicity, they will be ignored for now, but a better approach would be necessary in the short term.
3. Related to the previous one: if the message is heavily edited, should we consider that the suggestion was used? Again, for now the answer would be yes, but a smarter approach should be used.

### PR DESCRIPTION
**Pull request by faloi**
Closes #106

### CODE REVIEW

File: app/routes/messages.rb [hunk 1]
```diff
@@ -16,15 +16,27 @@ def assignment_query
     with_organization_and_course 'exercise.eid': json_body[:exercise_id], 'student.uid': json_body[:uid]
   end
 
+  def find_or_create_suggestion(assignment)
```
Comment (aguspina) [inline app/routes/messages.rb]: This should be a class method in Suggestion model

File: lib/classroom/models/assignment.rb [hunk 2]
```diff
@@ -49,6 +51,12 @@ def with_full_messages(user)
     self
   end
 
+  def add_message_to_submission!(message, sid)
+    submission = self.add_message! message, sid
```
Comment (aguspina) [inline lib/classroom/models/assignment.rb]: unnecessary self

File: app/routes/messages.rb (hunk 1 above)
Comment (faloi) [inline app/routes/messages.rb]: Mmm, I don't agree here. `suggestion_id` is a nullable field that can or cannot be included in the POST request. Why should the model be aware of this?

Comment (faloi) [inline app/routes/messages.rb]: :up: 
Please don't let this PR stall. It already has conflicts. :disappointed:

Comment (aguspina) [inline app/routes/messages.rb]: Dismiss my comment, please resolve conflicts!

**Comment by aguspina** [comment]
Thanks @faloi !!

**Pull request merged**